src/validation/fields/date.validation.ts

```diff
@@ -4,43 +4,40 @@ import {
   checkDateFieldMonth,
   checkDateFieldYear,
   checkDateOfBirth,
-  checkDateYearValueIsFourNumbers,
   checkIdentityDate,
   checkStartDate
 } from "../custom.validation";
 import { ErrorMessages } from "../error.messages";
 
 export const start_date_validations = [
-  body("start_date-day").custom((value, { req }) => checkDateFieldDay(ErrorMessages.DAY, req.body["start_date-day"], req.body["start_date-month"], req.body["start_date-year"])),
-  body("start_date-month").custom((value, { req }) => checkDateFieldMonth(ErrorMessages.MONTH, req.body["start_date-day"], req.body["start_date-month"], req.body["start_date-year"])),
-  body("start_date-year").custom((value, { req }) => checkDateFieldYear(ErrorMessages.YEAR, req.body["start_date-day"], req.body["start_date-month"], req.body["start_date-year"])),
-  body("start_date-year").custom((value, { req }) => checkDateYearValueIsFourNumbers(ErrorMessages.YEAR_LENGTH, req.body["start_date-year"])),
+  body("start_date")
```

Comment:
Most of this is a formatting change but referencing 'start-date-day' and not 'start-date' must have always been wrong (unless someone spots a hole in this change). In any case, this change (and others below) ensures that the inline error messages when parts of dates are missing appear correctly.

Reply:
Comment superseded by @markpit 's latest changes.